js/containers/portal-dashboard/portal-dashboard-app.tsx

```diff
@@ -91,7 +93,7 @@ function mapStateToProps(state: RootState) {
   };
 }
 
-const mapDispatchToProps = (dispatch: any, ownProps: any) => {
+const mapDispatchToProps = (dispatch: any, ownProps: any): Partial<IProps> => {
```

Comment:
The `Partial<IProps>` return type could also potentially be added to `mapStateToProps`, which exposes a very minor Typescript complaint about the type of `clazzName`. Not that important at all, but might be nice to add just as an example of writing a complete container component in TS, since we all like to copy from what has already been done elsewhere.

Reply:
Good thinking.  I made this change and added additional fixes as needed.

Reply:
Looks great. The other way to do it, if you didn't want the prop `clazzName` to be `string | false` would have been

```
function mapStateToProps(state: RootState): Partial<IProps> {
 ...
  return {
    isFetching: data.get("isFetching"),
    ...,
    clazzName: dataDownloaded ? state.getIn(["report", "clazzName"]) : undefined,
  };
}
```
but this is also totally fine.

Reply:
Yeah, I like this better.  I'm not crazy about or-ing in that boolean in the props.  Feels confusing.  Fixed.